src/Column/ColumnDefinitionParser.php

```diff
@@ -24,32 +24,50 @@ final class ColumnDefinitionParser extends \Yiisoft\Db\Syntax\ColumnDefinitionPa
         . '|\w*'
         . ')(?:\(([^)]+)\))?)(\[[\d\[\]]*\])?\s*/i';
 
-    public function parse(string $definition): array
+    protected function parseDefinition(string $definition): array
     {
         preg_match(self::TYPE_PATTERN, $definition, $matches);
 
         /** @var string $type */
         $type = $matches[3] ?? preg_replace('/\s*\(\d+\)/', '', $matches[1]);
-        $type = strtolower($type);
-        $info = ['type' => $type];
-
-        $typeDetails = $matches[4] ?? $matches[2] ?? '';
-
-        if ($typeDetails !== '') {
-            if ($type === 'enum') {
-                $info += $this->enumInfo($typeDetails);
-            } else {
-                $info += $this->sizeInfo($typeDetails);
-            }
-        }
 
-        if (isset($matches[5])) {
-            /** @psalm-var positive-int */
-            $info['dimension'] = substr_count($matches[5], '[');
-        }
-
-        $extra = substr($definition, strlen($matches[0]));
+        return [
+            $type,
+            $matches[4] ?? $matches[2] ?? null,
+            $matches[5] ?? null,
+            substr($definition, strlen($matches[0])),
+        ];
+    }
 
-        return $info + $this->extraInfo($extra);
+    protected function parseTypeParams(string $type, string $params): array
+    {
+        return match ($type) {
+            'bit varying',
+            'bit',
+            'bpchar',
+            'char',
+            'character varying',
+            'character',
+            'decimal',
+            'double precision',
+            'float4',
+            'float8',
+            'int',
+            'interval',
+            'numeric',
+            'real',
+            'string',
+            'time with time zone',
+            'time without time zone',
+            'time',
+            'timestamp with time zone',
+            'timestamp without time zone',
+            'timestamp',
+            'timestamptz',
+            'timetz',
+            'varbit',
+            'varchar' => $this->parseSizeInfo($params),
+            default => [],
+        };
```

Comment:
Abstract, pseudo, or user-defined types must also be parsable.

Reply:
Why need to parse abstract and pseudo type?  What case when this is needed?

Isn't custom types are support parameters?